packages/project/src/db/project.ts

```diff
@@ -84,35 +84,64 @@ export const create = async ({
   return project;
 };
 
-export const clone = async (clonableDomain: string, userId: string) => {
-  const clonableProject = await loadByDomain(clonableDomain);
-  if (clonableProject === null) {
-    throw new Error(`Not found project "${clonableDomain}"`);
+export const markAsDeleted = async (projectId: Project["id"]) => {
+  return await prisma.project.update({
+    where: { id: projectId },
+    data: { isDeleted: true },
+  });
+};
+
+export const rename = async (projectId: Project["id"], title: string) => {
+  if (title.length < MIN_TITLE_LENGTH) {
+    return { errors: `Minimum ${MIN_TITLE_LENGTH} characters required` };
   }
 
-  const prodBuild = await db.build.loadByProjectId(clonableProject.id, "prod");
+  return await prisma.project.update({
+    where: { id: projectId },
+    data: { title },
+  });
+};
 
-  if (prodBuild === undefined) {
-    throw new Error("Expected project to be published first");
-  }
+const clone = async (
+  originalProject: Project,
+  userId: string,
+  title?: string
+) => {
+  const build = await db.build.loadByProjectId(originalProject.id, "dev");
 
   const project = await prisma.$transaction(async (client) => {
     const project = await client.project.create({
       data: {
         userId,
-        title: clonableProject.title,
-        domain: generateDomain(clonableProject.title),
+        title: title ?? originalProject.title,
+        domain: generateDomain(originalProject.title),
       },
     });
 
-    await db.build.create(project.id, "dev", prodBuild, client);
+    await db.build.create(project.id, "dev", build, client);
 
     return project;
   });
 
   return project;
 };
 
+export const duplicate = async (projectId: string, userId: string) => {
+  const project = await loadById(projectId);
+  if (project === null) {
+    throw new Error(`Not found project "${projectId}"`);
+  }
+  return await clone(project, userId, `${project.title} (copy)`);
+};
+
+export const cloneByDomain = async (domain: string, userId: string) => {
+  const project = await loadByDomain(domain);
+  if (project === null) {
+    throw new Error(`Not found project "${domain}"`);
+  }
+  return await clone(project, userId);
+};
```

Comment:
We used to clone "prod" build in the old `clone()`, but in `cloneByDomain()` that replaces it we clone "dev" build.

Probably doesn't matter, just wanted to point out in case this is an accidental change we don't want.

Reply:
oh, def. accidental, forgotten during refactoring